### PR TITLE
Cmd line file fixes

### PR DESCRIPTION
**Pull request by dcbaker**
I noticed a couple of bugs in the wipe/reconfigure code, over the last couple of days. First I noticed that between 0.49 and now the code wouldn't function due to the removal of data structures from our internal representation. Then I noticed while trying to test a fix that of the coredata.dat file was empty (ie, created by touch) there would be an error. Then I noticed that native files are serialized into the cmd_line.txt either, when I was writing a script to convert cmd_line.txt into a command line that could be fed back to meson.

### CODE REVIEW

File: mesonbuild/environment.py [hunk 1]
```diff
@@ -332,7 +332,7 @@ def __init__(self, source_dir, build_dir, options):
             try:
                 self.coredata = coredata.load(self.get_build_dir())
                 self.first_invocation = False
-            except FileNotFoundError:
+            except (FileNotFoundError, EOFError):
```
Comment (xclaesse) [inline mesonbuild/environment.py]: Shouldn't try to recover from the cmd line file in that case?

File: mesonbuild/coredata.py [hunk 2]
```diff
@@ -576,6 +576,8 @@ def read_cmd_line_file(build_dir, options):
     properties = config['properties']
     if options.cross_file is None:
         options.cross_file = properties.get('cross_file', None)
+    if not options.native_file:
+        options.native_file = properties.get('native_file', [])
```
Comment (xclaesse) [inline mesonbuild/coredata.py]: I don't think this is going to deserialize into a list of strings. But rather to a string `'['/path1', '/path2']'`

Comment (dcbaker) [reply mesonbuild/coredata.py]: let me go look.

File: mesonbuild/environment.py (hunk 1 above)
Comment (dcbaker) [reply mesonbuild/environment.py]: I think we should. What's the difference between "my file was completely deleted" and "all of the content of my file was deleted?"

File: mesonbuild/coredata.py (hunk 2 above)
Comment (dcbaker) [reply mesonbuild/coredata.py]: You are correct, I'll fix that.

File: mesonbuild/environment.py (hunk 1 above)
Comment (xclaesse) [reply mesonbuild/environment.py]: FileNoteFoundError means we never configured the build dir, so it's a first configuration. Other errors means we configured but something is broken in the coredata file so we should read the cmd file and reconfigure from scratch.

Comment (xclaesse) [reply mesonbuild/environment.py]: To be more specific, I think coredata.load() should raise a MesonException if it gets EOFError

Comment (dcbaker) [reply mesonbuild/environment.py]: Okay, I  can move that to coredata.

**Comment by dcbaker** [comment]
@xclaesse, do the changes look good to you?

**Pull request merged**